internal/config/ociartifact/ociartifact.go

```diff
@@ -121,12 +160,17 @@ func (o *OCIArtifact) Pull(ctx context.Context, img string, opts *PullOptions) (
 		return nil, fmt.Errorf("read with limit: %w", err)
 	}
 
-	if err := verifyDigest(&layer, layerBytes); err != nil {
+	if err := verifyDigest(layer, layerBytes); err != nil {
 		return nil, fmt.Errorf("verify digest of layer: %w", err)
 	}
 
+	if err := o.impl.DatabaseWrite(db, key, layerBytes); err != nil {
```

Comment:
No way to leverage diskv's streaming API here?

Note: diskv's streaming API seems to be a bit buggy at the moment.

Reply:
We need the bytes in any case to validate the digests. That's why I thought we should not write anything to disk which is invalid.

Reply:
Done